### PR TITLE
Proposal to remove dependency on sigopt and wandb

### PR DESCRIPTION
**Pull request by lucasosouza**
Proposal to remove dependency on sigopt and wandb. Even though they are in requirements.txt, we should avoid importing them if they are not being used - that can avoid bugs within those libraries from breaking our code. Let me know what you think.

### CODE REVIEW

File: nupic/research/frameworks/vernon/parser_utils.py [hunk 1]
```diff
@@ -175,6 +174,9 @@ def process_args(args, config):
         insert_experiment_mixin(config, ray_wandb.WorkerLogger, prepend_name=False)
 
     if "create_sigopt" in args:
+
+        from nupic.research.frameworks.sigopt import SigOptExperiment
```
Comment (subutai) [inline nupic/research/frameworks/vernon/parser_utils.py:178]: This is not related to your change, but this parsing of the args doesn't look right. You should be able to run SigOptExperiment without creating a new experiment (e.g. you just want to run another SigOpt trial).

Comment (mvacaporale) [reply nupic/research/frameworks/vernon/parser_utils.py:178]: You can run sigopt without creating a new experiment. The handling of `create_sigopt` is solely for creating an experiment, as you noticed. But to run an experiment, you simply have to make sure you're using the `SigoptSurpervisedTrainable`. When calling via `run_with_raytune.run`, this equates to ensuring you set `ray_trainable=SigoptSurpervisedTrainable` or simply defining `sigopt_config` in your config which is used to interpret the proper default trainable.

**Review by mvacaporale — APPROVED** [review]

File: nupic/research/frameworks/vernon/parser_utils.py (hunk 1 above)
Comment (subutai) [reply nupic/research/frameworks/vernon/parser_utils.py:178]: OK. I have to say this is quite confusing. I looked through the code and couldn't figure it out. It's really confusing looking through all the parsers, and trying to figure out what's going on with the run command. It used to be so simple before!! It looks like you simply have to "know this" somehow.

**Review by subutai — APPROVED** [review]

File: nupic/research/frameworks/vernon/parser_utils.py [hunk 2]
```diff
@@ -151,6 +145,11 @@ def process_args(args, config):
 
     if "wandb" in args and args.wandb:
 
+        from wandb import util
+        from nupic.research.frameworks.wandb import ray_wandb
+        from nupic.research.frameworks.dynamic_sparse.common.ray_custom_loggers_2 \
```
Comment (lscheinkman) [inline nupic/research/frameworks/vernon/parser_utils.py:150]: What is the relationship between `nupic.research.frameworks.dynamic_sparse` and `wandb`?

Comment (lucasosouza) [reply nupic/research/frameworks/vernon/parser_utils.py:150]: That is a good point. The reason I moved that there is that the custom loggers are only used inside the `if wandb`. We should probably import those from someplace else other than dynamic sparse, I can put that in a separate PR. This one just moves the imports where they are used to reduce dependencies.

**Review by lscheinkman — APPROVED** [review]

File: nupic/research/frameworks/vernon/parser_utils.py (hunk 2 above)
Comment (lucasosouza) [reply nupic/research/frameworks/vernon/parser_utils.py]: Luiz I ended up fixing it here, it is a very small change.

**Pull request merged**